pmlc/dialect/stdx/transforms/i1_to_i32.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 Intel Corporation
+
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/IR/StandardTypes.h"
+#include "mlir/Pass/Pass.h"
+#include "pmlc/dialect/stdx/transforms/pass_detail.h"
+
+using namespace mlir; // NOLINT[build/namespaces]
+
+namespace pmlc::dialect::stdx {
+
+namespace {
+
+/// Changes loadOp from i1 memref to loadOp i32 followed by creating constant
+/// value 0 of i32 type and doing cmpi afterwards, next this bool-like type will
+/// be produced
+class LoadOpI1ToI32 final : public OpRewritePattern<LoadOp> {
+public:
+  using OpRewritePattern<LoadOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(LoadOp loadOp,
+                                PatternRewriter &rewriter) const override;
+};
+
+/// Changes storeOp to i1 memref to i32 select(val, 1, 0) followed by storeOp to
+/// i32
+class StoreOpI1ToI32 final : public OpRewritePattern<StoreOp> {
+public:
+  using OpRewritePattern<StoreOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(StoreOp storeOp,
+                                PatternRewriter &rewriter) const override;
+};
+} // namespace
+
+/// Changes loadOp from i1 memref to loadOp i32 followed by creating constant
+/// value 0 of i32 type and doing cmpi afterwards, next this bool-like type will
+/// be produced
+LogicalResult LoadOpI1ToI32::matchAndRewrite(LoadOp loadOp,
+                                             PatternRewriter &rewriter) const {
+  auto elementType = loadOp.result().getType();
+  if (!elementType.isInteger(1)) {
+    return failure();
+  }
+
+  auto destType = rewriter.getIntegerType(32);
+  auto loc = loadOp.getLoc();
+
+  loadOp.getMemRef().setType(
+      MemRefType::get(loadOp.getMemRefType().getShape(), destType));
```

Comment:
Same goes here, it seems like if we are using `replaceOp`, then we don't need to use `setType`